src/mailroom_madness.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+"""File contains a mailroom helper function."""
+from __future__ import division
+import sys
+
+
+# backwards compatible input
+try:
+    input = raw_input
+except NameError:
+    pass
+
+
+def prompt_donor_name(donor_data):
+    """Function gets name from stdin.
+
+    If name is list, list donor_data.
+    If name is not in donor_data, enter it with [] value.
+    Returns name.
+
+    Args:
+        donor_data (dict)
+    """
+    name = get_name(donor_data)
+    if not name:
+        return False
+    donor_data.setdefault(name, [])
+    return name
+
+
+def get_name(donor_data, out=sys.stdout):
+    """Get a name from the user.
+
+    Return the amount or false if the
+    user wishes to return to the main prompt.
+
+    Args:
+        donor_data (dict)
+        out (file, optional)
+    """
+    prompt = u'Enter a donor name, "list", or "cancel" > '
+    while True:
+        name = input(prompt)
```

Comment:
Notice how this call to `input` is intertwined with code that handles the result it gives.  This will make testing your functions hard and is a sign of code architecture that could be improved.